### ISSUE TITLE
Unable to access ToRecipients or CcRecipients

### ISSUE THREAD

**Issue by rousselle** [gautamsi/ews-javascript-api#69]
We were trying to get header information from the first (newest) message in a mailbox (https://gist.github.com/rousselle/d9fe1c07aa18587a14fbd3b5ac4a429d). We were unable to access the ToRecipients or the CcRecipients. We determined that this was due to a bug in ComplexPropertyCollection. I have attached a fix that seems to resolve the issue.

[ourfix.txt](https://github.com/gautamsi/ews-javascript-api/files/353086/ourfix.txt)

**Comment by gautamsi** [comment]
I am working on it. ComplexPropertyCollection is a base class which can not reference obj.Mailbox, we have separate things to check. 

I will fix it in a day or two.

**Comment by gautamsi** [comment]
it will be fixed in next update (0.6). you can rebuild by updating lines 123-126 like this in ComplexPropertyCollection.ts.

```ts
let collection: TComplexProperty[] = jsObjectCollection[this.GetCollectionItemXmlElementName()];
if (!ArrayHelper.isArray(collection)) {
     collection = [jsObjectCollection[this.GetCollectionItemXmlElementName()]];
}
```
or change ComplexPropertyCollection.js lines 136-139 (under node_modules\ews-javascript-api folder)

```js
var collection = jsObjectCollection[this.GetCollectionItemXmlElementName()];
if (!ExtensionMethods_1.ArrayHelper.isArray(collection)) {
     collection = [jsObjectCollection[this.GetCollectionItemXmlElementName()]];
}
```

`GetCollectionItemXmlElementName()` is an abstract method implemented by all child class of ComplexPropertyCollection

**Comment by gautamsi** [comment]
I need to fix it differently, it fails one way or other with given change. I fixed it for ExtendedPropertyCollection type, now it breaks other things.  I will try to make some changes which may fix it. it may take few days.

**Comment by rousselle** [comment]
Do you have an ETA for 0.6 ?

**Comment by rousselle** [comment]
Thanks. That would be very helpful

**Comment by gautamsi** [comment]
published 0.5.1 on npm. let me know how it work.

**Comment by rousselle** [comment]
:+1: 0.5.1 looks good. We have everything we need now. Thank you very much for all your help. It's been a pleasure working with you.
